src/javascript/app/base/client.js

```diff
@@ -59,6 +59,38 @@ const Client = (() => {
                 applyToAllElements('.is-logout', (el) => {
                     el.style.display = 'none';
                 });
+                
+                // Only hide skeleton loaders if not on a loading trading page and header is ready
+                const trading_init_progress = getElementById('trading_init_progress');
+                const is_trading_loading = trading_init_progress && trading_init_progress.style.display !== 'none';
+                
+                // Check if header is ready via global window object to avoid circular dependency
+                let is_header_ready = false;
+                if (typeof window !== 'undefined' && window.Header && window.Header.isHeaderReady) {
+                    is_header_ready = window.Header.isHeaderReady();
+                } else {
+                    // If header readiness check not available, assume ready
+                    is_header_ready = true;
+                }
+                
+                if (!is_trading_loading && is_header_ready) {
+                    // Hide skeleton loaders for logged-in state (login buttons not needed)
+                    applyToAllElements('.skeleton-btn-login, .skeleton-btn-signup', (el) => {
+                        el.style.display = 'none';
```

Comment:
Instead of inline style manipulation, can we use css classes for show/hide logic?
```suggestion
                        el.classList.add('hidden');
```
where
```
.hidden { display: none; }
```